lib/index.ts

```diff
@@ -0,0 +1,70 @@
+import { Server, Plugin } from "@hapi/hapi";
+import * as HapiAuthJwt from "hapi-auth-jwt2";
+import verifyJWT, { Validator } from "./verifyJWT";
+import * as keystoresFromFile from "./keystores.json";
+import { assert } from "@hapi/hoek";
+import makeIssueTokenRoute from "./make-issue-token-route";
+
+export type ClientSecrets = {
+  [key: string]: string;
+};
+
+type HapiOidcOptions = {
+  tokenEndpoint?: string;
+  clients?: ClientSecrets;
+  fetchKeystore?: () => object | Promise<object>;
+  validate?: Validator;
+  dev?: boolean;
+  omitCheckExp?: boolean;
+};
+
+const logger = (server: Server) => (
+  tags: string | Array<string>,
+  message?: object | string,
+  timestamp?: number
+) => server.log(tags, message, timestamp);
+
+type AuthorizationGrantTokenPayload = {
+  readonly client_id: string;
+};
+
+type PasswordGrantTokenPayload = {
+  readonly client_id: string;
+  readonly username: string;
+  readonly password: string;
+  readonly grant_type: "password";
+};
```

Comment:
I'll want to flesh out these types to help document the different types of grants. Probably soon on 2.0 but not on this PR